src/main/java/org/jenkinsci/test/acceptance/po/ToolInstallation.java

```diff
@@ -80,21 +80,21 @@ private static boolean hasUpdatesFor(final Jenkins jenkins, Class<? extends Tool
         final Control button = page.control(by.button("Add " + name));
 
         String pathPrefix = button.resolve().getAttribute("path").replaceAll("repeatable-add", "tool");
-        String path = page.createPageArea(pathPrefix, new Runnable() {
-            @Override public void run() {
-                button.click();
-            }
-        });
+        String path = page.createPageArea(pathPrefix, button::click);
         return page.newInstance(type, jenkins, path);
     }
 
     public static <T extends ToolInstallation> void installTool(Jenkins jenkins, Class<T> type, String name, String version) {
         waitForUpdates(jenkins, type);
 
+        jenkins.getPluginManager().checkForUpdates();
+        
         ConfigurablePageObject tools = ensureConfigPage(jenkins);
         T maven = addTool(jenkins, type);
         maven.name.set(name);
-        maven.installVersion(version);
+        if(version != null) {
+            maven.installVersion(version);
+        }
```

Comment:
This is actually not your code but "maven" variable name is somewhat misleading.